nncf/common/utils/backend.py

```diff
@@ -85,7 +91,10 @@ def is_tensorflow_model(model: TModel) -> bool:
     """
     import tensorflow  # type: ignore
 
-    return isinstance(model, tensorflow.Module)
+    try:
+        return isinstance(model, tensorflow.Module)
```

Comment:
I installed the latest `tensorflow` and `tensorflow.Module` is present there. I think `get_available_backends` is not working properly.

This fix is approved but I would suggest to check what nncf is reported in the env of bug about supported frameworks during `import nncf`. The output on my desktop:
```
INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, tensorflow, onnx, openvino 
```
If NNCF reports tensorflow then you should add fix in this part as well.

Reply:
> If NNCF reports tensorflow then you should add fix in this part as well.

NNCF reports that tensorflow exists even if we place an empty tensorflow.py file into PYTHONPATH. How can we fix this? Could we verify each needed tensorflow part on our side to provide correct report?

Reply:
What do you think about adding a model entity import check to the backend get function?

Reply:
I don't get it, sorry. Did you mean to verify that model object may be imported from the package here - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/common/utils/backend.py#L29-L51?

Reply:
@KodiaqQ  Would using https://docs.python.org/3.9/library/importlib.metadata.html#distributions instead of `import_module` inside `get_available_backends` help here?


```python
# pip install torch
# my_torch.py - script in cwd 

from importlib import metadata, import_module
import_module("torch")
# <module 'torch' from '/home/adokucha/work/nncf/venv/lib/python3.10/site-packages/torch/__init__.py'>
import_module("my_torch")
#<module 'my_torch' from '/home/adokucha/work/nncf/my_torch.py'>
metadata.distribution("torch")
# <importlib.metadata.PathDistribution object at 0x7b3d0bd537f0>
metadata.distribution("my_torch")
#   raise PackageNotFoundError(name)
#importlib.metadata.PackageNotFoundError: No package metadata was found for my_torch
```

Reply:
@AlexanderDokuchaev will this work if I install a package that builds locally?

Reply:
> I don't get it, sorry. Did you mean to verify that model object may be imported from the package here - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/common/utils/backend.py#L29-L51?

Yes, that's what I meant.

Reply:
@alexsu52, @AlexanderDokuchaev I've updated the proposal a little, but this message will still be here.

> INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, tensorflow, onnx, openvino

Reply:
> @alexsu52, @AlexanderDokuchaev I've updated the proposal a little, but this message will still be here.
> 
> > INFO:nncf:NNCF initialized successfully. Supported frameworks detected: torch, tensorflow, onnx, openvino

Do you have suggestions on how to fix this?

Reply:
```python
frameworks = [
    ("torch", BackendType.TORCH),
    ("torch", BackendType.TORCH_FX),
    ("tensorflow", BackendType.TENSORFLOW),
    ("onnx", BackendType.ONNX),
    ("openvino", BackendType.OPENVINO),
]

try:
    distribution(module_name)
except ...
```

Think that it will be enough.
`distribution` check installed packages by name, not namespaces of modules
`fx` is part of `torch`, i dont know easy way to broke the check (without creating fake package with same name). 
For openvino looks same, but if possible to install openvino package without runtime, we can combine `distribution(module_name.split('.')[0]` and `import_module(module_name)` checks

Reply:
> Do you have suggestions on how to fix this?

No. From the user's perspective, the presence of the `tensorflow` looks correct, even as an empty file.
Another approach is to remove the notion of the supported frameworks.

Reply:
@alexsu52, @AlexanderDokuchaev, was a final solution found? How should I update this PR? Do we need it?

Reply:
I agree with removing the message about supported frameworks during import if you are not able to implement it correctly.

Reply:
Agreed that this issue will be fixed in the follow-up PR.